util/predicates/cluster_predicates.go

```diff
@@ -188,3 +188,27 @@ func ClusterUnpausedAndInfrastructureReady(logger logr.Logger) predicate.Funcs {
 	// Use any to ensure we process either create or update events we care about
 	return Any(log, createPredicates, updatePredicates)
 }
+
+// ClusterIsManagedTopology returns a Predicate that returns true when cluster.Spec.Topology
+// is NOT nil and false otherwise.
+func ClusterIsManagedTopology(logger logr.Logger) predicate.Funcs {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			log := logger.WithValues("predicate", "ClusterIsManagedTopology", "eventType", "update")
+			cluster := &clusterv1.Cluster{}
+
+			log.WithValues("namespace", cluster.Namespace, "cluster", cluster.Name)
+
+			if cluster.Spec.Topology != nil {
+				log.V(6).Info("Cluster has topology, allowing reconcile")
+				return true
+			}
+
+			log.V(4).Info("Cluster does not have topology, blocking further reconcile")
```

Comment:
nit: Can we use the working `allowing further processing` and `blocking further processing` (here and in l.203). Just for consistency with other predicates